tests/unit/transformer/test_mapper.py

```diff
@@ -299,3 +309,87 @@ def test_ambiguous_column_names_exception_is_ignored(self, input_df, mapping, ex
         transformer = Mapper(mapping, ignore_ambiguous_columns=True)
         output_df = transformer.transform(input_df)
         assert_df_equality(expected_output_df, output_df)
+
+
+class TestValidateDataTypes:
+    @pytest.fixture(scope="class")
+    def input_df(self, spark_session):
+        return spark_session.createDataFrame(
+            [Row(col_a=123, col_b="Hello", col_c=123456789)],
+            schema="col_a int, col_b string, col_c long"
+        )
+
+    @pytest.fixture(scope="class")
+    def matching_mapping(self) -> List[tuple]:
+        # fmt: off
+        return [
+            ("col_a", "col_a", T.IntegerType()),
+            ("col_b", "col_b", T.StringType()),
+            ("col_c", "col_c", T.LongType()),
+        ]
+        # fmt: on
+
+    def test_matching_mapping_with_casting(self, input_df, matching_mapping):
```

Comment:
data types match so this should not trigger any casting, right? maybe `test_matching_mapping_without_validation`. Some use `_with_validation` suffix and some not. As all except two tests apply validation would remove the suffix as the class name already indicates that it is about validation.